### PR TITLE
Require annotations option

### PR DESCRIPTION
**Pull request by matttattoli**
Implemented annotations selector on pods similarly to the required labels selector. I went ahead and updated the docs and option test. Let me know what you think or if anything should be changed.

### CODE REVIEW

File: reaper/options.go [hunk 1]
```diff
@@ -112,6 +115,24 @@ func labelRequirement() (*labels.Requirement, error) {
...
+		return nil, fmt.Errorf("specified %s but not %s", envRequireAnnotationKey, envRequireAnnotationValues)
+	} else if !annotationKeyExists && annotationValuesExist {
+		return nil, fmt.Errorf("did not specify %s but did specify %s", envRequireAnnotationKey, envRequireAnnotationValues)
+	} else if !annotationKeyExists && !annotationValuesExist {
+		return nil, nil
+	}
+	annotationValues := strings.Split(annotationValue, ",")
+	annotationRequirement, err := labels.NewRequirement(annotationKey, selection.In, annotationValues)
+	if err != nil {
+		return nil, fmt.Errorf("could not create requirement annotation: %s", err)
```
Comment (ojagodzinski) [inline reaper/options.go:131]: *annotation requirement

File: reaper/reaper.go [hunk 2]
```diff
@@ -63,6 +63,18 @@ func (reaper reaper) getPods() *v1.PodList {
 		logrus.WithError(err).Panic("unable to get pods from the cluster")
 		panic(err)
 	}
+	filteredList := []v1.Pod{}
```
Comment (ojagodzinski) [inline reaper/reaper.go:66]: should be moved inside if statement

**Comment by matttattoli** [comment]
Hey, thanks for the review. I fixed everything you commented on.

**Comment by brianberzins** [comment]
This is weird case to me... but for a really specific reason that I remember running into about 3 years ago when working on a platform style project within kubernetes that wanted to use some form of selecting pods. 

I explicitly remember digging into the differences between labels and annotations, and in the case of what pod-reaper is doing, labels were clearly the way to go. They're used to efficiently run selection queries on the API. In this case, we we'd be running a larger (more broad) query against the API and then filtering out pods based on the pods annotations. In short, I'm pretty confident that labels are the way to go... HOWEVER...

I also remember running into situations with a certain orchestration tool (Spinnaker) that didn't really allow users full control over pod labels and annotations. It's been a long while since I've looked at spinnaker, so I'm not sure if they've actually addressed that or not, but there are likely other tools out there that are abstracting away people's abilities to define what they want in pod layouts.

So in short, my take is: labels are better for this than annotations, but I understand there are situations where this is going to work and the performance hit is pretty unlikely to matter to almost anyone.

**Comment by brianberzins** [comment]
For transparency -- looks like spinnaker made the label issue I commented on 3 years ago obsolete around 2 years ago with v2:

https://github.com/spinnaker/spinnaker/issues/1624#issuecomment-308165818

**Comment by matttattoli** [comment]
Well I hope this PR doesn't become obsolete 🙂 

I don't see too much of an issue keeping it as an option if users need it. Guess it's still up to you if you want to merge it or not.

**Comment by brianberzins** [comment]
> I don't see too much of an issue keeping it as an option if users need it. Guess it's still up to you if you want to merge it or not.

I've got you! I'm on it :)

**Comment by brianberzins** [comment]
Release 2.8.0 when everything clears through CI/Docker builds
